### PR TITLE
Junit, timeout etc.

### PR DESCRIPTION
**Pull request by rebinskiy**
Added junit library, timeout and some functionality to client.js

Is it maybe better to create a docker image with junit already inside and use that one?

Needs speed optimisation and stress testing.

### CODE REVIEW

**Comment by VincenzoFerme** [comment]
@Rebinskiy can you explain me how do you do it now, and what is the difference compared to the workflow you follow when you do not have dependencies of your code for third-party libraries?

**Comment by rebinskiy** [comment]
@VincenzoFerme now I just pass a locally stored library to container.putArchive() when initialising a container that need to use the library, the idea is that the client should give information about needed libraries, by default now is junit only

**Comment by VincenzoFerme** [comment]
@StKyr please upload here the image and the details out of our discussion. @Rebinskiy discuss with @StKyr about how to approach the problem. Moreover you should also discuss with @triglian about how to pipe directly the input from the user to the Docker copy command.

**Comment by VincenzoFerme** [comment]
Great you decided to give it a try. In case of any issue ping me on this pull request.

File: app.js [hunk 1]
```diff
@@ -8,101 +8,35 @@ const server = require('./server');
 const javaBox = require('./javaBox');
 
 
-server.on('runJava', runSingleClass);
+server.on('runJava', runJava_2);
 javaBox.on('result', giveFeedBack);
 
 
-function runSingleClass(clientId, main, files, timeLimitCompile, timeLimitExecution) {
+function runJava_2(clientId, main, files, timeLimitCompile, timeLimitExecution) {
```
Comment (VincenzoFerme) [inline app.js:15]: Change the name to a more meaningful one.

**Review by VincenzoFerme — CHANGES_REQUESTED** [review]

File: client.js [hunk 2]
```diff
@@ -117,8 +117,8 @@ public class AllTests
 }`
                 }
             ]},
-            timeLimitCompile : 10000,
-            timeLimitExecution : 10000
+            compileTimeoutMs : 1000000,
```
Comment (VincenzoFerme) [inline client.js:120]: Why exactly 16 mins and 40 seconds?

**Review by VincenzoFerme — CHANGES_REQUESTED** [review]

File: javaBox.js [hunk 3]
```diff
@@ -75,15 +76,40 @@ function dockerCommand(command, commandTimeLimit, nextCommand) {
     const execution = (err, container) => {
 
         if (!err) container.exec(opts, (err, exec) => {
-            exec.start((err, stream) => waitCmdExit(container, exec, nextCommand, stream, commandTimeLimit));
+            exec.start((err, stream) => {
```
Comment (VincenzoFerme) [inline javaBox.js:79]: You should be able to obtain the complete log from the container at the end of its execution, before deleting it. But this solution is fine as well.

**Review by VincenzoFerme — APPROVED** [review]

**Pull request merged**